thoth/messaging/message_base.py

```diff
@@ -55,12 +56,14 @@ def __init__(
         bootstrap_server: str = "localhost:9092",
         topic_retention_time_second: int = 60 * 60 * 24 * 45,
         protocol: str = "SSL",
+        message_version: int = 0,
     ):
         """Create general message."""
         topic_prefix = os.getenv("THOTH_DEPLOYMENT_NAME", None)
         self.topic_name = topic_name or "thoth.base-topic"
         if topic_prefix is not None:
             self.topic_name = f"{topic_prefix}.{self.topic_name}"
+        self.topic_name = f"{self.topic_name}.v{self._base_version}.{message_version}"
```

Comment:
Do we want to have topics per message version? As our deployments should always produce messages of one version per message type, I think we could keep topic names version agnostic. Could be easier for consumers to subscribe and parse version supported in their side.

Reply:
I think topics per message version is a good way of doing this, I think it makes it obvious in terms of what order to upgrade things. After updating message then update producer and then once all current messages have been consumed update consumer.  If we make it part of the message contents then the consumer will have to come up with logic on what to do if it receives a message for a version that it might not support and either throw the message out or it will have to constantly move the message to the back of the queue until the consumer has been updated.

Reply:
OK, sounds reasonable :+1: +1